Bluepill-cli/Bluepill-cli/Bluepill/Bluepill.m

```diff
@@ -308,6 +308,20 @@ - (void)checkProcessWithContext:(BPExecutionContext *)context {
         return;
     }
 
+    // This check should be last after all of the more specific tests
+    // It checks if the app is even running, which it must be at this point
+    // If it's not running and we passed the above checks (e.g., the tests are not yet completed)
+    // then it must mean the app has crashed.
+    // However, we have a short-circuit for tests because those may not actually run any app
+    if (!isRunning && context.pid > 0 && !self.config.testing_NoAppWillRun) {
```

Comment:
The `testing_NoAppWillRun` variable is for the parsing tests right? I would then maybe also use it in the `kill()` below where I am just looking for `self.appPID == 0` for the same case.